lib/server.js

```diff
@@ -210,7 +210,16 @@ class Server {
 
         if (err instanceof errors.FloraError) {
             if (err.data) response.data = err.data;
-            if (err.meta) Object.assign(response.meta, err.meta);
+            if (err.meta && typeof err.meta === 'object') {
+                if (typeof err.meta.headers === 'object') {
+                    for (let header in err.meta.headers) {
+                        response.meta.headers[header] = err.meta.headers[header];
+                    }
+                }
```

Comment:
The response object has as setter method for headers:
```suggestion
                if (typeof err.meta.headers === 'object') {
                    for (let [header, value] of Object.entries(err.meta.headers || {})) {
                        response.header(header, value);
                    }
                }
```
```Object.entries``` throws a ```TypeError``` on null values, so we need to use a fallback (nullish coalescing operator is not supported in configured ECMAScript version).

I'm not sure this version is more readable in the end :man_shrugging: 

Functional version just for fun :wink: 
```js
Object.entries(err.meta.headers || {}).forEach(([header, value]) => response.header(header, value));
```